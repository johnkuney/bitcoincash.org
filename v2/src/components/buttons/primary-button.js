import primBtnStyles from "./primary-button.module.css"
import React from "react"
import { useStaticQuery, graphql } from "gatsby"
import Link from "../../global/link"

export const PurePrimaryButton = ({ data, buttonText, href, noMarginLeft }) => {
  const theme = data.site.siteMetadata.themeColours
  return (
    <Link to={href}>
      <button
        className={`${noMarginLeft ? primBtnStyles.noMarginLeft : ""} ${
          primBtnStyles.primBtn
        }`}
        style={{
          backgroundImage: `linear-gradient(270deg, ${theme.secondary_light} 0%, ${theme.secondary_dark} 100%)`,
        }}
      >
        {buttonText}
      </button>
    </Link>
  )
}

const PrimaryButton = props => {
  const data = useStaticQuery(graphql`
    query PrimaryButtonThemeQuery {
      site {
        siteMetadata {
          themeColours {
            primary_dark
            primary_light
            secondary_dark
            secondary_light
          }
        }
      }
    }
  `)
  return <PurePrimaryButton {...props} data={data} />
}

export default PrimaryButton
