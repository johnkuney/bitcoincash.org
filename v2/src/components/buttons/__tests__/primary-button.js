import React from "react"
import renderer from "react-test-renderer"
import { PurePrimaryButton } from "../primary-button"

describe("PrimaryButton", () => {
  it("renders correctly", () => {
    const data = {
      site: {
        siteMetadata: {
          themeColours: {
            primary_dark: "#040c3c",
            primary_light: "#212c6e",
            secondary_dark: "#bb5a00",
            secondary_light: "#ff8d00",
          },
        },
      },
    }

    const tree = renderer
      .create(<PurePrimaryButton data={data} href="http://localhost" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
