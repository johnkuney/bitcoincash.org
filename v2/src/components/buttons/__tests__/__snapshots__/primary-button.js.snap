// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton renders correctly 1`] = `
<a
  href="http://localhost"
>
  <button
    className=" primBtn"
    style={
      Object {
        "backgroundImage": "linear-gradient(270deg, #ff8d00 0%, #bb5a00 100%)",
      }
    }
  />
</a>
`;
